### ISSUE TITLE
[@types/next-auth] Custom Provider

### ISSUE THREAD

**Issue by sebascomeau** [DefinitelyTyped/DefinitelyTyped#51980]
Is something wrong with CI or publishing infrastructure?

- Check the [Infrastructure status updates](https://github.com/DefinitelyTyped/DefinitelyTyped/issues/44317) thread for known issues.
- Leave a message on the [DefinitelyTyped Gitter](http://gitter.im/DefinitelyTyped/DefinitelyTyped).

If you know how to fix the issue, make a pull request instead.

- [x] I tried using the `@types/next-auth` package and had problems.
- [x] I tried using the latest stable version of tsc. https://www.npmjs.com/package/typescript
- [x] I understand that `@types` package versions [do not follow SemVer](https://github.com/DefinitelyTyped/DefinitelyTyped#how-do-definitely-typed-package-versions-relate-to-versions-of-the-corresponding-library).
- [x] I have a question that is inappropriate for [StackOverflow](https://stackoverflow.com/).  (Please ask any appropriate questions there).
- [x] [Mention](https://github.com/blog/821-mention-somebody-they-re-notified) the authors (see `Definitions by:` in `index.d.ts`) so they can respond.
  - Authors: @luia @iaincollins @JuanM04 @balazsorban44 @euxn23

I need a way to add a custom provider that doesn't exists with Next-Auth. With version of @types/next-auth 3.7.2 there is no way to do this.

```ts
export interface DefaultProviders {
    Apple: Apple;
    Attlassian: Atlassian;
    Auth0: Auth0;
    AzureADB2C: AzureADB2C;
    Basecamp: Basecamp;
    BattleNet: BattleNet;
    Box: Box;
    Bungie: Bungie;
    Cognito: Cognito;
    Credentials: Credentials;
    Discord: Discord;
    Email: Email;
    EVEOnline: EVEOnline;
    Facebook: Facebook;
    Foursquare: Foursquare;
    FusionAuth: FusionAuth;
    GitHub: GitHub;
    GitLab: GitLab;
    Google: Google;
    IdentityServer4: IdentityServer4;
    LINE: LINE;
    LinkedIn: LinkedIn;
    MailRu: MailRu;
    Medium: Medium;
    Netlify: Netlify;
    Okta: Okta;
    Reddit: Reddit;
    Salesforce: Salesforce;
    Slack: Slack;
    Spotify: Spotify;
    Strava: Strava;
    Twitch: Twitch;
    Twitter: Twitter;
    VK: VK;
    Yandex: Yandex;
}

export type Providers = Array<ReturnType<DefaultProviders[keyof DefaultProviders]>>;

export interface NextAuthOptions {
    providers: Providers;
    ...
}
```

I am only allowed to used one of the DefaultProviders. I need to add my own provider like Keycloak or add one with a different name than the one provided.

**Comment by euxn23** [comment]
@sebascomeau 
Thank you for reporting this issue. This is a degrade, and I fixed it in #52007 .
I'm afraid to have made degrade, but please wait merge and the release of new version.

**Comment by martin-badin** [comment]
@sebascomeau Please, you can close this issue if it is fixed.
